sdks/java/io/redis/src/main/java/org/apache/beam/sdk/io/redis/RedisIO.java

```diff
@@ -0,0 +1,294 @@
+ * Optionally, you can provide a key pattern (to filter the keys). The following example
+ * illustrates how to configure a source:
+ *
+ * <pre>{@code
+ *
+ *  pipeline.apply(RedisIO.read()
+ *    .withConnection(RedisConnection.create(Collections.singletonList("localhost:6379")))
+ *    .withKeyPattern("foo*")
+ *
+ * }</pre>
+ */
+public class RedisIO {
+
+  private static final Logger LOG = LoggerFactory.getLogger(RedisIO.class);
+
+  public static Read read() {
+    return new  AutoValue_RedisIO_Read.Builder()
+        .setCommand(Read.Command.GET).setKeyPattern("*").build();
+  }
+
+  private RedisIO() {
+  }
+
+  /**
+   * A {@link PTransform} reading key/value pairs from a Redis database.
+   */
+  @AutoValue
+  public abstract static class Read extends PTransform<PBegin, PCollection<KV<String, String>>> {
+
+    /**
+     * The Redis commands related to read of key-value pairs.
+     */
+    public enum Command {
+      GET
+    }
+
+    @Nullable abstract RedisConnection connection();
+    @Nullable abstract Command command();
+    @Nullable abstract String keyPattern();
+    @Nullable abstract RedisService redisService();
+
+    abstract Builder builder();
+
+    @AutoValue.Builder
+    abstract static class Builder {
+      @Nullable abstract Builder setConnection(RedisConnection connection);
+      @Nullable abstract Builder setCommand(Command command);
+      @Nullable abstract Builder setKeyPattern(String keyPattern);
+      @Nullable abstract Builder setRedisService(RedisService redisService);
+      abstract Read build();
+    }
+
+    /**
+     * Define the connection to the Redis server.
+     *
+     * @param connection The {@link RedisConnection}.
+     * @return The corresponding {@link Read} {@link PTransform}.
+     */
+    public Read withConnection(RedisConnection connection) {
+      checkArgument(connection != null, "RedisIO.read().withConnection(connection) called with "
+          + "null connection");
+      return builder().setConnection(connection).build();
+    }
+
+    /**
+     * Define the Redis command to execute to retrieve the key/value pairs.
+     *
+     * @param command The Redis command.
+     * @return The corresponding {@link Read} {@link PTransform}.
+     */
+    public Read withCommand(Command command) {
+      checkArgument(command != null, "RedisIO.read().withCommand(command) called with null "
+          + "command");
+      return builder().setCommand(command).build();
+    }
+
+    /**
+     * Define the pattern to filter the Redis keys.
+     * @param keyPattern The filter key pattern.
+     * @return The corresponding {@link Read} {@link PTransform}.
+     */
+    public Read withKeyPattern(String keyPattern) {
+      checkArgument(keyPattern != null, "RedisIO.read().withKeyPattern(pattern) called with "
+          + "null pattern");
+      return builder().setKeyPattern(keyPattern).build();
+    }
+
+    /**
+     * Define the Redis service to use.
```

Comment:
This comment didn't help me understand what this method is for and how/when to use it.